apps/agile-project-fe/src/app/components/input/input.tsx

```diff
@@ -1,17 +1,33 @@
 import styled from 'styled-components';
 
 export interface InputProps {
-  variant: 'email';
+  variant: 'text' | 'email' | 'password';
 }
 
-const StyledInput = styled.div`
-  color: pink;
-`;
+const StyledInput = styled.div``;
+
+export function Input({ variant }: InputProps) {
+  const inputType =
+    variant === 'text'
+      ? 'text'
+      : variant === 'email'
+      ? 'email'
+      : variant === 'password'
+      ? 'password'
+      : 'default';
+
+  const inputPlaceholder =
+    variant === 'text'
+      ? 'Digite o texto aqui'
+      : variant === 'email'
+      ? 'Digite o email'
+      : variant === 'password'
+      ? 'Digite a senha'
+      : 'default';
```

Comment:
These ternary operators chained together are too hard to read and maintain, please use either an object literal mapping like so:

```ts
const placeholder: Record<'text' | 'email' | 'password', string> = {
    text:  "Digite o texto aqui",
    // continua...
}
```